src/main/java/uk/gov/cshr/controller/signup/SignupController.java

```diff
@@ -126,30 +142,42 @@ public String sendInvite(Model model,
     }
 
     @GetMapping("/{code}")
-    public String signup(Model model, @PathVariable(value = "code") String code) {
-        if (inviteService.isInviteValid(code)) {
-            Invite invite = inviteRepository.findByCode(code);
-
-            if (!invite.isAuthorisedInvite()) {
-                log.debug("Invite email = {} not yet authorised - redirecting to enter token screen", invite.getForEmail());
-                return REDIRECT_ENTER_TOKEN + code;
-            }
+    public String signup(Model model, @PathVariable(value = "code") String code,
+                                            RedirectAttributes redirectAttributes) {
+        if (inviteService.isCodeExists(code)) {
+            if (!inviteService.isCodeExpired(code)) {
```

Comment:
Could this if condition to made positive by switching the "else" part?

Reply:
Done